doc/architecture-decisions/0002-add-enhanced-ecommerce.md

```diff
@@ -0,0 +1,46 @@
+# 2. Add Enhanced E-commerce plugin
+
+Date: 2017-06-27
+
+## Status
+
+Accepted
+
+## Context
+
+Rendering apps track pageviews and events to help us measure the performance
+of GOV.UK.
+
+Static uses an [abstraction layer](https://github.com/alphagov/govuk_frontend_toolkit/blob/master/docs/analytics.md) in GOV.UK Frontend Toolkit, that reports the data to Google Analytics, but the underlying provider could be changed.
+
+For analysing search performance, we want to be able to reliably track clickthrough rates on links in search results.
+
+[Enhanced E-Commerce](https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce) is a plugin for google analytics that can track impressions as well as clicks. We model each search result page as a "product list", and each content item as a "product".
+
+## Decision
+
+We added a plugin for E-Commerce to Static, so we could try using it
+to measure search performance.
+
+We didn't add an abstraction layer to GOV.UK Frontend Toolkit, because we don't
```

Comment:
IIRC, the Frontend team would have preferred us to implement it in Frontend Toolkit. And the reason we didn't do it like that (as well as your reasons below) was that it made it technically challenging to ensure the GA events all fired in the correct order. Is that right, @brenetic?